subsystem/kubeapi_test.go

```diff
@@ -73,6 +80,7 @@ func deployPullSecretResource(ctx context.Context, client k8sclient.Client, name
 }
 
 func deployClusterDeploymentCRD(ctx context.Context, client k8sclient.Client, spec *hivev1.ClusterDeploymentSpec) {
+	deployClusterImageSetCRD(ctx, client, spec.Provisioning.ImageSetRef)
```

Comment:
Maybe we could have some negative flows?
What is expected if the ImageSet does not exists, or if it is created after the ClusterDeployment?

Reply:
The ImageSet is mandatory for creating a cluster (e.g. as pull secret), so createNewCluster would fail if missing and Reconcile is retried. Are the tests in kubeapi_test should check such scenarios?

Reply:
Added a test for creating a ClusterDeployment with a missing ClusterImageSet resource (i.e. creating the deployment first).